### DISCUSSION TITLE
Custom VPC Network Configuration

### DISCUSSION THREAD

**Discussion by shawnlovelidge** [aws/aws-cdk#31185] [Q&A]
It should be possible to create your own custom Vpc() (i.e a new Vpc() with NO subnetConfiguration)

If I try to add my own subnet infrastructure and associate the subnets to the vpc programatically via the use of CfnSubnet() & CfnSubnetRouteTableAssociation() respectively then whenever I try to iterate through subnets via (vpc.publicSubnets()) I get this error.

" Error: There are no 'Public' subnet groups in this VPC. Available types:..."

It at almost seems like if a developer doesn't use new vpc(..., { subnetConfiguration[] }) aws-cdk doesn't recognize additional created subnets. WHY? This just seems off.... Why have an aws-sdk if we can't use the api to do custom network configurations?

What am I missing? Is there a way to configure a network and have the underlying api know about your configuration?

**Comment by khushail** [comment]
@github-lernender-corp , Could you please share minimal repro code that is producing the error. That would be helpful to see what might be causing the error.
Thanks
